src/Text/Pandoc/Readers/LaTeX.hs

```diff
@@ -726,6 +728,14 @@ dolstinline = do
 doLHSverb :: PandocMonad m => LP m Inlines
 doLHSverb = codeWith ("",["haskell"],[]) <$> manyTill (satisfy (/='\n')) (char '|')
 
+dosiunitx :: PandocMonad m => LP m Inlines
+dosiunitx = do
+  skipopts
+  value <- char '{' >> manyTill anyChar (char '}')
+  preunit <- (char '[' >> manyTill anyChar (char ']')) <|> (return "")
+  unit <- char '{' >> manyTill anyChar (char '}')
+  pure <$> str $ preunit ++ (if length preunit == 0 then "" else "\160") ++ value ++ (if length unit == 0 then "" else "\160") ++ unit
```

Comment:
This is fine, if only a bit too long for the 80 characters limit that's usually used.

An alternative would be

``` haskell
return . str . mconcat $ [preunit, value, unit]  -- plus the optional spaces
```

Getting the full length of the string is a bit wasteful, as we only need to know if it's empty.  `null` can be used instead.